File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        levels=[]
        if not root:
            return []
        currLevel=[root]
        nextLevel=[]
        while currLevel: 
            # Push Child Node to next level
            for n in currLevel:
                if(n.left):
                    nextLevel.append(n.left)
                if(n.right):
                    nextLevel.append(n.right)

            levels.append([n.val for n in currLevel])
            currLevel=nextLevel
            nextLevel=[]
        return levels 
            

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levels;
    vector<vector<int>> levelOrder(TreeNode* root) {
        if (root == NULL)
            return {};
        vector<TreeNode*> currLevel;
        vector<TreeNode*> nextLevel;
        currLevel.push_back(root);

        vector<int> nodeValueCurrLevel;
        while (currLevel.size() != 0) {
            nodeValueCurrLevel={};
            for (TreeNode* n : currLevel) {
                nodeValueCurrLevel.push_back(n->val);
                if (n->left)
                    nextLevel.push_back(n->left);
                if (n->right)
                    nextLevel.push_back(n->right);
            }
            levels.push_back(nodeValueCurrLevel);
            currLevel = nextLevel;
            nextLevel = {};
        }
        return levels;
    }
};